### ISSUE TITLE
Exit Code 139 on random CLI xplat runs

### ISSUE THREAD

**Issue by eerhardt** [dotnet/coreclr#5672]
Lately I've been seeing more and more "Expected command to pass but it did not. Exit Code: 139" issues on CLI runs.

In the past, when we've seen "Exit Code 139", it means "segmentation fault" and usually in crossgened managed code.

I don't have a solid repro yet.

Opening this issue to track all the places we've seen it. Maybe from logging these failures, and looking through the logs we will be able to determine what is the exact cause of the Exit Code 139.

Here are the first logs:

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0-preview2/job/debug_debian8.2_x64_prtest/97/testReport/junit/Microsoft.DotNet.Tests/PackagedCommandTests/TestProjectDependencyIsNotAvailableThroughDriver/

```
Expected command to pass but it did not. File Name: /mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/debug_debian8.2_x64_prtest/artifacts/debian.8-x64/stage2/dotnet Arguments: --verbose build --build-profile "/mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/debug_debian8.2_x64_prtest/test/dotnet.Tests/bin/Debug/netcoreapp1.0/TestAssets/TestProjects/AppWithDirectDependency/project.json" Exit Code: 139 StdOut: StdErr:
```

https://mseng.visualstudio.com/dotnetcore/_build?favDefinitionId=3598&_a=summary&buildId=3052704

```
2016-06-09T17:36:40.6997100Z     Microsoft.DotNet.Tools.Builder.Tests.BuildPerformanceTest.IncrementalSkipAllNoDependenciesInGraph_TwoTargetGraphLarge [FAIL]
2016-06-09T17:36:40.7002620Z       Expected command to pass but it did not.
2016-06-09T17:36:40.7008200Z       File Name: /opt/code/artifacts/centos.7-x64/stage2/dotnet
2016-06-09T17:36:40.7016130Z       Arguments: --verbose build  --no-dependencies  "/opt/code/test/Performance/bin/Release/netcoreapp1.0/IncrementalSkipAllNoDependenciesInGraph_TwoTargetGraphLarge/PerformanceTestProjects/TwoTargetGraphLarge/TwoTargetLargeP4"    --framework netcoreapp1.0           
2016-06-09T17:36:40.7022120Z       Exit Code: 139
2016-06-09T17:36:40.7026560Z       StdOut:
2016-06-09T17:36:40.7030930Z       
2016-06-09T17:36:40.7036240Z       StdErr:
```

https://mseng.visualstudio.com/dotnetcore/_build?favDefinitionId=3603&_a=summary&buildId=3053398

```
2016-06-09T21:08:52.706Z: Executing - /Users/SHAREDADMIN/vsoagent/_work/6/s/artifacts/osx.10.11-x64/stage2/dotnet --verbose build --build-profile   "/Users/SHAREDADMIN/vsoagent/_work/6/s/artifacts/osx.10.11-x64/tests/artifacts/5c513543-b685-4730-88ad-47d72b6ddccb/e2etestroot/project.json" -o "/Users/SHAREDADMIN/vsoagent/_work/6/s/artifacts/osx.10.11-x64/tests/artifacts/5c513543-b685-4730-88ad-47d72b6ddccb/e2etestroot/test space/bin"   --framework netcoreapp1.0           
2016-06-09T21:08:52.830Z: Telemetry is: Enabled
2016-06-09T21:08:53.682Z:     Microsoft.DotNet.Tests.EndToEnd.EndToEndTest.TestDotnetBuild [FAIL]
2016-06-09T21:08:53.682Z:       Expected command to pass but it did not.
2016-06-09T21:08:53.682Z:       File Name: /Users/SHAREDADMIN/vsoagent/_work/6/s/artifacts/osx.10.11-x64/stage2/dotnet
2016-06-09T21:08:53.682Z:       Arguments: --verbose build --build-profile   "/Users/SHAREDADMIN/vsoagent/_work/6/s/artifacts/osx.10.11-x64/tests/artifacts/5c513543-b685-4730-88ad-47d72b6ddccb/e2etestroot/project.json" -o "/Users/SHAREDADMIN/vsoagent/_work/6/s/artifacts/osx.10.11-x64/tests/artifacts/5c513543-b685-4730-88ad-47d72b6ddccb/e2etestroot/test space/bin"   --framework netcoreapp1.0           
2016-06-09T21:08:53.682Z:       Exit Code: 138
```

Technically that one is exit code **138**, but this is OSX so that could be the difference.

https://mseng.visualstudio.com/dotnetcore/_build?favDefinitionId=3605&_a=summary&buildId=3052708

```
2016-06-09T17:35:55.829Z: Build failed: Microsoft.DotNet.Cli.Build.Framework.BuildFailureException: Command failed with exit code 139: /opt/code/artifacts/ubuntu.14.04-x64/stage2/dotnet "build" "--configuration" "Release"
2016-06-09T17:35:55.829Z: 
2016-06-09T17:35:55.829Z:    at Microsoft.DotNet.Cli.Build.TestTargets.BuildTests(BuildTargetContext c)
2016-06-09T17:35:55.829Z:    at Microsoft.DotNet.Cli.Build.Framework.BuildContext.ExecTarget(BuildTarget target)
```

@brthor @piotrpMSFT @gkhanna79 @sergiy-k @JohnChen0 @adityamandaleeka @joshfree

**Comment by stephentoub** [comment]
> Technically that one is exit code 138, but this is OSX so that could be the difference.

138 is likely a SIGBUS

**Comment by gkhanna79** [comment]
One of the cases @sergiy-k and I looked with @brthor was with standalone apps that only have IL and no crossgen.

That said, ti will be good to get logs or repros to look into this further.

**Comment by eerhardt** [comment]
From my quick scan of the above failures, they all failed during `dotnet build`, which is all crossgen'd (shared framework and 'sdk' code, both of which is crossgen'd).

**Comment by gkhanna79** [comment]
@janvorli do you have any thoughts on this?

**Comment by gkhanna79** [comment]
Can we consider enabling core dump generation on OSX (http://stackoverflow.com/questions/9412156/how-to-generate-core-dumps-in-mac-os-x) and Ubuntu (https://wiki.ubuntu.com/Apport) on the machines running CLI builds? This will enable us to have dumps captured that we can then examine to determine what is really going on.

**Comment by eerhardt** [comment]
Here's some more failures:

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0-preview2/job/release_osx_x64_prtest/114/console

```
01:23:20 EC       >]   /Users/dotnet-bot/j/workspace/dotnet_cli/rel_1.0.0-preview2/release_osx_x64_prtest/artifacts/osx.10.11-x64/stage2/dotnet "build" "--framework" "netcoreapp1.0"
01:23:21 EC       <] IL]  /Users/dotnet-bot/j/workspace/dotnet_cli/rel_1.0.0-preview2/release_osx_x64_prtest/TestAssets/TestProjects/PerformanceTestProjects/SingleTargetGraph/SingleTargetP2> /Users/dotnet-bot/j/workspace/dotnet_cli/rel_1.0.0-preview2/release_osx_x64_prtest/artifacts/osx.10.11-x64/stage2/dotnet "build" "--framework" "netcoreapp1.0" exited with 138
```

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0-preview2/job/release_rhel7.2_x64_prtest/110/console

```
01:34:49     Microsoft.DotNet.Tools.Builder.Tests.BuildPerformanceTest.BuildAllNoDependenciesInGraph_TwoTargetGraph [FAIL]
01:34:49       Expected command to pass but it did not.
01:34:49       File Name: /mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/release_rhel7.2_x64_prtest/artifacts/rhel.7.2-x64/stage2/dotnet
01:34:49       Arguments: --verbose build  --no-dependencies  "/mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/release_rhel7.2_x64_prtest/test/Performance/bin/Release/netcoreapp1.0/BuildAllNoDependenciesInGraph_TwoTargetGraph/PerformanceTestProjects/TwoTargetGraph/TwoTargetP1"    --framework netcoreapp1.0           
01:34:49       Exit Code: 139
```

**Comment by eerhardt** [comment]
@mmitche - What do you think about @gkhanna79's suggestion:

> Can we consider enabling core dump generation on OSX (http://stackoverflow.com/questions/9412156/how-to-generate-core-dumps-in-mac-os-x) and Ubuntu (https://wiki.ubuntu.com/Apport) on the machines running CLI builds? This will enable us to have dumps captured that we can then examine to determine what is really going on.

Note - CLI uses Docker to run on all the Linux distros, in case that makes any difference here.

**Comment by janvorli** [comment]
@gkhanna79 I think I have an idea that would help us figure out crashes in the lab without polluting the lab machines disks with core dumps. 
What if we let the core be generated, then ran lldb in batch mode to log the stack trace and then deleted the core? We could then see the logged stack trace in the console log from the lab.
To run lldb this way, you would do it as follows:
`lldb-3.8.1 -b -o "plugin load libsosplugin.so" -o "clrstack -f"  --core ./core ./corerun`
lldb executes the commands given using the `-o` options and then exits.
@mmitche does it sound like something we could do?

**Comment by gkhanna79** [comment]
@janvorli I think that sounds reaosnable. My suggestion was to start enabling infra pieces that can help us get more data than just failed logs.

**Comment by gkhanna79** [comment]
But SOS does not exist on OSX, right?

**Comment by janvorli** [comment]
It does exist there, we build it during coreclr build.

**Comment by eerhardt** [comment]
More occurrences in @toddm's PR:

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0-preview2/job/release_osx_x64_prtest/116/console

```
08:02:35 EC       <] IL]  > /Users/dotnet-bot/j/workspace/dotnet_cli/rel_1.0.0-preview2/release_osx_x64_prtest/artifacts/osx.10.11-x64/stage2/dotnet "pack" "/Users/dotnet-bot/j/workspace/dotnet_cli/rel_1.0.0-preview2/release_osx_x64_prtest/src/Microsoft.DotNet.Compiler.Common/project.json" "--no-build" "--build-base-path" "/Users/dotnet-bot/j/workspace/dotnet_cli/rel_1.0.0-preview2/release_osx_x64_prtest/artifacts/osx.10.11-x64/tests/packagesBuild" "--output" "/Users/dotnet-bot/j/workspace/dotnet_cli/rel_1.0.0-preview2/release_osx_x64_prtest/artifacts/osx.10.11-x64/tests/packages" "--version-suffix" "003025" exited with 138
```

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0-preview2/job/debug_ubuntu16.04_x64_prtest/117/testReport/junit/Microsoft.Dotnet.Tools.Test.Tests/GivenThatWeWantToRunTestsForMultipleTFMsInTheConsole/It_builds_and_runs_tests_for_netcoreapp10/

```
Expected command to pass but it did not. File Name: /mnt/j/workspace/workspace/dotnet_cli/rel_1.0.0-preview2/debug_ubuntu16.04_x64_prtest/artifacts/ubuntu.16.04-x64/stage2/dotnet Arguments: test /mnt/j/workspace/workspace/dotnet_cli/rel_1.0.0-preview2/debug_ubuntu16.04_x64_prtest/test/dotnet-test.Tests/bin/Debug/netcoreapp1.0/.ctorConsoleTests/ProjectsWithTests/MultipleFrameworkProject/project.json -f netcoreapp1.0 Exit Code: 139 StdOut: StdErr:
```

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0-preview2/job/debug_fedora23_x64_prtest/111/consoleFull

```
08:26:34 EC       <] IL]  > /mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/debug_fedora23_x64_prtest/artifacts/fedora.23-x64/stage2/dotnet "pack" "/mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/debug_fedora23_x64_prtest/src/Microsoft.DotNet.ProjectModel/project.json" "--no-build" "--build-base-path" "/mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/debug_fedora23_x64_prtest/artifacts/fedora.23-x64/stage2compilation/forPackaging" "--output" "/mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/debug_fedora23_x64_prtest/artifacts/fedora.23-x64/packages" "--configuration" "Debug" "--version-suffix" "003025" exited with 134
```

That last one is Exit Code **134** on Fedora - I haven't seen that one before.

**Comment by eerhardt** [comment]
More occurrences on my latest CLI PR:

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0-preview2/job/debug_opensuse13.2_x64_prtest/110/

```
Expected command to pass but it did not. File Name: /mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/debug_opensuse13.2_x64_prtest/artifacts/opensuse.13.2-x64/stage2/dotnet Arguments: --verbose build --build-profile "/mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/debug_opensuse13.2_x64_prtest/test/dotnet-build.Tests/bin/Debug/netcoreapp1.0/AppDefaultPaths4/TestAppWithLibrary/TestApp/project.json" -o "/mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/debug_opensuse13.2_x64_prtest/test/dotnet-build.Tests/bin/Debug/netcoreapp1.0/AppDefaultPaths4/TestAppWithLibrary/out" -b /mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/debug_opensuse13.2_x64_prtest/test/dotnet-build.Tests/bin/Debug/netcoreapp1.0/AppDefaultPaths4/TestAppWithLibrary/build --framework netcoreapp1.0 Exit Code: 139
```

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0-preview2/job/release_ubuntu_x64_prtest/120/

```
Expected command to pass but it did not. File Name: /opt/code/artifacts/ubuntu.14.04-x64/stage2/dotnet Arguments: publish /opt/code/TestAssets/TestProjects/DependencyContextValidator/TestApp/project.json Exit Code: 139 StdOut: StdErr:
```

i think that this error is occurring during `dotnet build`. I see `dotnet publish` and `dotnet pack` fail in recently, but publish and pack call build. I haven't seen any other dotnet commands failing.  The problem is it fails building different projects, I haven't seen the same project twice.

The most troubling thing about these 2 failures is that [this PR](https://github.com/dotnet/cli/pull/3493) is built using the absolute latest dotnet/coreclr release/1.0.0 build:  `rc3-24210-01`.  So this `Exit Code 139` issue during `dotnet build` is still occurring with the latest coreclr and crossgen bits.

**Comment by eerhardt** [comment]
Maybe we don't need to change Jenkins in anyway. To try to repro this locally, we could install the latest CLI on a Linux machine, and just start `dotnet build`ing a bunch of different projects in a loop and wait for it to fail.

**Comment by janvorli** [comment]
If you can repro it locally, that would be ideal.

**Comment by gkhanna79** [comment]
>  just start dotnet building a bunch of different projects in a loop and wait for it to fail.

@eerhardt What kind of projects? Would looping in building HelloWorld from dotnet new suffice or something?

**Comment by eerhardt** [comment]
> What kind of projects? Would looping in building HelloWorld from dotnet new suffice or something?

I would start with all the projects under:

https://github.com/dotnet/cli/tree/rel/1.0.0-preview2/TestAssets/TestProjects

Those are all our "test projects" in the CLI.  The ones under "PerformanceTestProjects" are pretty good at repro'ing the issue it seems. I see three of the logs above using test projects under that repo.

Steps:

1. Install the latest CLI: https://github.com/dotnet/cli/tree/rel/1.0.0-preview2#installers-and-binaries
 - I'd use the .tar.gz ".NET Core SDK Binaries" link on the far right of the table.  Unzip it to a directory on your path.
2. `git clone https://github.com/dotnet/cli.git`
 - `git checkout -b origin/rel/1.0.0-preview2`
3. `dotnet restore $REPO_ROOT/TestAssets/TestProjects`
3. `dotnet build $REPO_ROOT/TestAssets/TestProjects/**/project.json`

That will try to build all the test projects in that directory.  If that doesn't fail, try looping on all the projects, or maybe just the projects under "PerformanceTestProjects". Sorry there isn't a solid repro, but looking through the logs, it seems `dotnet build` may be the key.

**Comment by janvorli** [comment]
@gkhanna79 sure, I'll to do that.

**Comment by gkhanna79** [comment]
@eerhardt I am getting the following error when trying step 3 above:

```
Errors in TestAssets/TestProjects/DependencyContextFromTool/project.json
    Unable to resolve 'dotnet-dependency-context-test (>= 1.0.0)' for '.NETCoreApp,Version=v1.0'.

Errors in TestAssets/TestProjects/AppWithDirectDependencyWithOutputName/project.json
    Unable to resolve 'dotnet-dependency-tool-invoker (>= 1.0.0)' for '.NETCoreApp,Version=v1.0'.
```

Any ideas?

**Comment by eerhardt** [comment]
yes, sorry - the easiest thing you can do to unblock yourself is delete those 2 project.json files on your local disk.

I forgot that there are NuGet packages that get built during the other parts of the build, and then some of our TestProjects depend on those packages.  Removing those TestProjects in your clone will let you continue.

**Comment by gkhanna79** [comment]
Thanks - that worked.

**Comment by eerhardt** [comment]
@gkhanna79 - @pranavkm just picked up the CLI from this morning and he said his Linux build failed twice in a row with "exit code 134".

Maybe sync with him - if he can get a more solid repro.

**Comment by gkhanna79** [comment]
@pranavkm Can you please share more details about the repro?

**Comment by stephentoub** [comment]
(For reference, that exit code translates to a SIGABRT.)

**Comment by pranavkm** [comment]
Our build log had this:

```
Exec
   program: dotnet
   commandline: pack --no-build src/Microsoft.Extensions.Globalization.CultureInfoCache --configuration Release
   workingdir: /home/aspnetagent/TeamCity/buildAgent/work/29495d0a71cf35ac/Universe/Localization

fail: Exit code 134 from dotnet
```

`dotnet build` on that project had already succeeded at this point. This on is a Ubuntu 14.04 machine with CLI version 1.0.0-preview2-003024

**Comment by pranavkm** [comment]
It passed the third time around. So it's not super consistent.

**Comment by eerhardt** [comment]
Here's another case:

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0-preview2/job/debug_opensuse13.2_x64_prtest/116/console

```
15:57:52 Compiling dotnet-compile.UnitTests for .NETCoreApp,Version=v1.0
15:57:52 EC       <] IL]  /mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/debug_opensuse13.2_x64_prtest/test/dotnet-compile.UnitTests> /mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/debug_opensuse13.2_x64_prtest/artifacts/opensuse.13.2-x64/stage2/dotnet "build" "--configuration" "Debug" exited with 139
```

**Comment by brthor** [comment]
I'm seeing this more frequently:

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0-preview2/job/debug_opensuse13.2_x64_prtest/120

**Comment by eerhardt** [comment]
Here are a couple more:

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0-preview2/job/release_osx_x64_prtest/128/testReport/junit/Microsoft.DotNet.Tools.Builder.Tests/IncrementalTests/TestRebuildChangedLockFile/

```
Expected command to pass but it did not. File Name: /Users/dotnet-bot/j/workspace/dotnet_cli/rel_1.0.0-preview2/release_osx_x64_prtest/artifacts/osx.10.11-x64/stage2/dotnet Arguments: --verbose build --build-profile "/Users/dotnet-bot/j/workspace/dotnet_cli/rel_1.0.0-preview2/release_osx_x64_prtest/test/dotnet-build.Tests/bin/Release/netcoreapp1.0/CreateTestInstance/TestSimpleIncrementalApp/project.json" -o "/Users/dotnet-bot/j/workspace/dotnet_cli/rel_1.0.0-preview2/release_osx_x64_prtest/test/dotnet-build.Tests/bin/Release/netcoreapp1.0/CreateTestInstance/TestSimpleIncrementalApp/bin" --framework netcoreapp1.0 Exit Code: 138 StdOut: StdErr:
```

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0-preview2/job/release_ubuntu_x64_prtest/130/console

```
17:56:27 EC       >]   /opt/code/artifacts/ubuntu.14.04-x64/stage2/dotnet "build" "-f" "netstandard1.6" "--build-base-path" "/opt/code/artifacts/ubuntu.14.04-x64/tests/packagesBuild" "/opt/code/src/Microsoft.DotNet.Files"
17:56:27 [39;49mEC       <] IL]  > /opt/code/artifacts/ubuntu.14.04-x64/stage2/dotnet "build" "-f" "netstandard1.6" "--build-base-path" "/opt/code/artifacts/ubuntu.14.04-x64/tests/packagesBuild" "/opt/code/src/Microsoft.DotNet.Files" exited with 139
```

**Comment by eerhardt** [comment]
If we can't get a repro on this with the steps I used above - one difference may be that our Jenkins and VSO jobs are all running on Ubuntu machines, but using Docker.  I just thought of that - all our CI jobs use Ubuntu as the base OS, and then use Docker to switch to OpenSUSE, Fedora, CentOS.

However, OSX doesn't use Docker - so that may rule this whole theory out.

**Comment by pranavkm** [comment]
One more OSX failure although this time it was during a test run:

```
Unhandled Exception: System.ObjectDisposedException: Safe handle has been closed
    at System.Runtime.InteropServices.SafeHandle.DangerousAddRef(Boolean& success)
    at System.StubHelpers.StubHelpers.SafeHandleAddRef(SafeHandle pHandle, Boolean& success)
    at Microsoft.Win32.Win32Native.SetEvent(SafeWaitHandle handle)
    at Microsoft.DotNet.Watcher.Tools.FunctionalTests.FileWatcherTests.<>c__DisplayClass8_1.<MultipleTriggers>b__1(Object _, String f)
    at System.IO.FileSystemWatcher.NotifyFileSystemEventArgs(WatcherChangeTypes changeType, String name)
    at System.IO.FileSystemWatcher.RunningInstance.FileSystemEventCallback(IntPtr streamRef, IntPtr clientCallBackInfo, IntPtr numEvents, String[] eventPaths, FSEventStreamEventFlags[] eventFlags, UInt64[] eventIds)
    at Interop.RunLoop.CFRunLoopRun()
    at System.IO.FileSystemWatcher.RunningInstance.WatchForFileSystemEventsThreadStart(Object arg)
    at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
fail: Exit code 134 from dotnet
fail: Stack trace: 
   at Viewf6b4e9d349294322a1489012268f5710.Dotnet (System.String command, System.String dotnetDir) <0x3309370 + 0x0065b> in <filename unknown>:0 
   at Viewf6b4e9d349294322a1489012268f5710.DotnetTest (System.String projectFile, System.String configuration, System.String test_options) <0x3308da8 + 0x0025b> in <filename unknown>:0
```

**Comment by stephentoub** [comment]
> One more OSX failure although this time it was during a test run

This one looks like it may be a bug in your test.  Some lambda in the MultipleTriggers function in the FileWatcherTests class is calling Set on a ManualResetEvent or an AutoResetEvent.

**Comment by eerhardt** [comment]
One more on Ubuntu 16

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0-preview2/job/debug_ubuntu16.04_x64_prtest/138/testReport/junit/Microsoft.Extensions.DependencyModel/FunctionalTests/PublishTest_appname___TestAppPortableDeps___portable__True__checkCompilation__False_/

```
Expected command to pass but it did not. File Name: /mnt/j/workspace/workspace/dotnet_cli/rel_1.0.0-preview2/debug_ubuntu16.04_x64_prtest/artifacts/ubuntu.16.04-x64/stage2/dotnet Arguments: publish /mnt/j/workspace/workspace/dotnet_cli/rel_1.0.0-preview2/debug_ubuntu16.04_x64_prtest/TestAssets/TestProjects/DependencyContextValidator/TestAppPortableDeps/project.json Exit Code: 139 StdOut: StdErr:
```

**Comment by eerhardt** [comment]
And another in our official VSO buld:

https://mseng.visualstudio.com/dotnetcore/_build?_a=summary&favDefinitionId=3605&buildId=3057535

```
2016-06-11T19:21:33.731Z:     Microsoft.DotNet.Tools.Builder.Tests.IncrementalTests.TestNoLockFile [FAIL]
2016-06-11T19:21:33.731Z:       Expected command to pass but it did not.
2016-06-11T19:21:33.731Z:       File Name: /opt/code/artifacts/ubuntu.14.04-x64/stage2/dotnet
2016-06-11T19:21:33.731Z:       Arguments: --verbose build --build-profile   "/opt/code/test/dotnet-build.Tests/bin/Release/netcoreapp1.0/CreateTestInstance/TestSimpleIncrementalApp/project.json" -o "/opt/code/test/dotnet-build.Tests/bin/Release/netcoreapp1.0/CreateTestInstance/TestSimpleIncrementalApp/bin"   --framework netcoreapp1.0           
2016-06-11T19:21:33.731Z:       Exit Code: 139
```

**Comment by janvorli** [comment]
Fixed by #5770

**Comment by sergiy-k** [comment]
Fixed by https://github.com/dotnet/coreclr/pull/5771 in the 1.0 release branch.
